iati/resources.py

```diff
@@ -241,125 +299,159 @@ def create_lib_data_path(name):
+    _ensure_portable_filepath(name)
+
     return path_for_version(os.path.join(PATH_SCHEMAS, '{0}'.format(name) + FILE_SCHEMA_EXTENSION), version)
 
 
+@iati.version.allow_possible_version
 @iati.version.normalise_decimals
 def folder_name_for_version(version=iati.version.STANDARD_VERSION_ANY):
     """Return the folder name for a given version of the Standard.
 
     Args:
-        version (str): The version of the Standard to return the folder path for. Defaults to iati.version.STANDARD_VERSION_ANY.
+        version (str / int / Decimal / iati.Version): The version of the Standard to return the folder path for. Defaults to iati.version.STANDARD_VERSION_ANY.
 
     Returns:
         str: The folder name for the specified version of the Standard.
 
     Raises:
-        ValueError: When a specified version is not a valid version of the IATI Standard.
-
-    Todo:
-        Extract magic string: 'version-independent'
+        TypeError: When a specified version is of a type that cannot represent an IATI version number.
+        ValueError: When a specified version is not a known version of the IATI Standard.
 
     """
+    is_major = False
+
     if version == iati.version.STANDARD_VERSION_ANY:
-        return 'version-independent'
-    elif isinstance(version, str):
+        return PATH_VERSION_INDEPENDENT
+    elif isinstance(version, (str, int)) and not isinstance(version, bool):
         try:
-            if int(version) in iati.version.STANDARD_VERSIONS_MAJOR:
-                return version
+            if str(int(version)) == str(version):
+                is_major = True
         except ValueError:
             pass
 
-        version = iati.Version(version)
+    if is_major and int(version) in iati.version.STANDARD_VERSIONS_MAJOR:
+        return str(version)
+    elif not is_major:
+        # is an iati.Version due to the `normalise_decimals` decorator
+        version.patch = 0  # folder names do not currently account for patch versions
 
-    try:
         if version in iati.version.STANDARD_VERSIONS:
             return str(version.integer) + '-0' + str(version.decimal)
-        elif version.major in iati.version.STANDARD_VERSIONS_MAJOR:
-            return str(version.major)
-    except AttributeError:
-        pass
 
-    raise ValueError("Version {} is not a valid version of the IATI Standard.".format(version))
+    raise ValueError("Version {0} is not a known version of the IATI Standard.".format(version))
 
 
 def folder_path_for_version(version=iati.version.STANDARD_VERSION_ANY):
     """Return the path for the folder containing SSOT data (schemas, codelists etc) at the specified version of the Standard.
 
     Args:
-        version (str): The version of the Standard to return the folder path for. Defaults to iati.version.STANDARD_VERSION_ANY.
+        version (str / int / Decimal / iati.Version): The version of the Standard to return the folder path for. Defaults to iati.version.STANDARD_VERSION_ANY.
```

Comment:
Should it still have a default?

Reply:
I'd say probably not...

Reply:
37cca6d69f7346e7af8ca448ab766c2efc051972